pkg/testutils/anvil/anvil.go

```diff
@@ -0,0 +1,76 @@
+package anvil
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"net"
+	"os"
+	"os/exec"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"github.com/xmtp/xmtpd/pkg/blockchain"
+)
+
+func findFreePort() (int, error) {
+	ln, err := net.Listen("tcp", "127.0.0.1:0")
+	if err != nil {
+		return 0, err
+	}
```

Comment:
server_test has exactly the same function called `getNextOpenPort` can we unify them?

Reply:
Good catch. Will use the other one